bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/BigtableIOHBase.java

```diff
@@ -0,0 +1,204 @@
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.client.Delete;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.client.Scan;
+
+import com.google.bigtable.v2.MutateRowsRequest;
+import com.google.bigtable.v2.Row;
+import com.google.bigtable.v2.RowFilter;
+import com.google.cloud.bigtable.config.BigtableOptions;
+import com.google.cloud.bigtable.hbase.BigtableConfiguration;
+import com.google.cloud.bigtable.hbase.BigtableOptionsFactory;
+import com.google.cloud.bigtable.hbase.adapters.Adapters;
+import com.google.cloud.bigtable.hbase.adapters.HBaseMutationAdapter;
+import com.google.cloud.bigtable.hbase.adapters.PutAdapter;
+import com.google.cloud.bigtable.hbase.adapters.read.DefaultReadHooks;
+import com.google.cloud.bigtable.util.ZeroCopyByteStringUtil;
+import com.google.common.base.Preconditions;
+import com.google.protobuf.ByteString;
+
+/**
+ * Utilities to configure {@link BigtableIO} with HBase artifacts
+ */
+@Experimental
+public class BigtableIOHBase {
+
+  public static Read read(String projectId, String instanceId, String tableId) {
+    return new Read(projectId, instanceId, tableId);
+  }
+
+  public static Write write(String projectId, String instanceId, String tableId) {
+    return new Write(projectId, instanceId, tableId);
+  }
+
+  private BigtableIOHBase() {}
+
+  /**
+   * This is a wrapper around {@link org.apache.beam.sdk.io.gcp.bigtable.BigtableIO.Read} that adds
+   * HBase semantics
+   */
+  public static class Read extends PTransform<PBegin, PCollection<Result>> {
+    private static final long serialVersionUID = 1L;
+
+    private static final DoFn<Row, Result> ROW_TO_RESULT_TRANSFORM = new DoFn<Row, Result>() {
+      private static final long serialVersionUID = 1L;
+      @ProcessElement
+      public void processElement(ProcessContext c) throws Exception {
+        c.output(Adapters.ROW_ADAPTER.adaptResponse(c.element()));
+      }
+    };
+
+    private final SerializableConfiguration serializableConfiguration;
+    private final String tableId;
+    private final SerializableScan scan = new SerializableScan(new Scan());
+
+    private Read(String projectId, String instanceId, String tableId) {
+      Preconditions.checkNotNull(projectId, "Please configure a projectId.");
+      Preconditions.checkNotNull(instanceId, "Please configure a instanceId.");
+      Preconditions.checkNotNull(tableId, "Please configure a tableId.");
+      this.serializableConfiguration =
+          new SerializableConfiguration(BigtableConfiguration.configure(projectId, instanceId));
+      this.tableId = tableId;
+    }
+
+    /**
+     * Add an extended configuration option.  See {@link BigtableOptionsFactory} for more details.
+     */
+    public Read withConfiguration(String key, String value) {
+      serializableConfiguration.get().set(key, value);
```

Comment:
Styleguide suggests similar,
Here is a rough sketch of using AutoValue to make this more consistent with Beam
https://github.com/sduskis/cloud-bigtable-client/pull/1